Polyfill amp-analytics script

adekbadek opened the pull request:
### All Submissions:

* [X] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Polyfill of `amp-analytics` AMP script (just the bits used by the plugin), part of #193.

A part that needs some explanation here is the linker parameter. Here's [a short overview](https://amp.dev/documentation/components/amp-analytics/#linkers), and here the [documentation about linker param receiving](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/linker-id-receiving.md) – the part that's actually polyfilled. Linker param _setting_ is not handled here, because it would be set only by a page served from AMP cache, which would be… an AMP page (not a version with polyfill).

Here's an example of a linker param:
```
?ref_newspack_cid=1*10zr7ce*cid*T2ptZDdJNU9EYndELWV0TlM1N0FwSGVwNHE3S240VkVVU0o3YlNsaFc2YVVPRXJhSWFhaVlsOTgtQXNsc21Eeg
```

The `T2pt…` part is base64-encoded client ID (of value `Ojmd7I5ODbwD-etNS57ApHep4q7Kn4VEUSJ7bSlhW6aUOEraIaaiYl98-AslsmDz`). You can this set the client ID to a specified value (in JS, e.g. `btoa('hello-world')` gives you a base64 encoded `hello-world`).

Note that the checksum part of the linker param is disregarded in this polyfill. I think the security implications of client ID spoofing in this use case are negligible.

### How to test the changes in this Pull Request:

1. Linker parameter
    1. Append a linker param to a post URL in non-AMP mode, visit the site in a fresh session (with no `newspack-cid` cookie set)
    2. Observe in the devtools that the `newspack-cid` cookie is set accordingly
    3. Observe that the linker param is stripped from the URL afterwards
    4. Confirm that this works in the same way on an AMP site – you can use [this AMP-cached site](https://campaigns--testing-newspackstaging-com.cdn.ampproject.org/c/s/campaigns-testing.newspackstaging.com/) to "generate" linker params with valid checksums
2. Reporting views to API 
    1. Create a campaign with "once" frequency
    2. Visit the site in a fresh session in non-AMP mode, observe the campaign is displayed (make sure you "see" it – that means at least 500ms with 90% of the campaign visible in the viewport)
    3. Reload the site or navigate to another post – observe the campaign is not displayed

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (approved):
This works as described and the polyfill is quite impressive. The new JS is pretty complex—do you think it's worth adding some JSDoc for the new functions? Otherwise, looks good to me and great job.

Tested in Chrome, Safari and Firefox and the linker functionality works as expected across non-AMP to AMP and vice versa.

adekbadek commented:
> The new JS is pretty complex—do you think it's worth adding some JSDoc for the new functions?

Good idea, added in effcfdc34a5a3ab99c45526dd7d695fe4369f597

Also, I've added handling of `ini-load` event to account for changes introduced in #305 in 6c43ab39a25bd81d6308d88d42c8fda4e2c0dc49

dkoo reviewed (approved):
:shipit:
